src/processor/operator/persistent/parquet_column_writer.cpp

```diff
@@ -0,0 +1,252 @@
+#include "processor/operator/persistent/parquet_column_writer.h"
+
+#include "processor/operator/persistent/csv_parquet_writer.h"
+
+using namespace kuzu::common;
+
+namespace kuzu {
+namespace processor {
+
+void ParquetColumnWriter::nextParquetColumn(LogicalTypeID logicalTypeID) {
+    switch (logicalTypeID) {
+    case LogicalTypeID::BOOL:
+        boolWriter = static_cast<parquet::BoolWriter*>(rowWriter->column(currentParquetColumn));
```

Comment:
Ditto. I think we should only keep one columnWriter pointer, instead of keeping one for each data type. Dynamic casting is fine here.
```
auto columnWriter = rowGroupWriter->column(currentParquetColumn)
    switch (logicalTypeID) {
    case LogicalTypeID::BOOL:
        auto boolWriter = static_cast<parquet::BoolWriter*>(columnWriter);
... ...
```

Reply:
I can only see the need of having to static_cast every time. Can this cause overhead?